scripts/core/menu/authoring-switch.ts

```diff
@@ -0,0 +1,45 @@
+import {authoringReactEnabledUserSelection, extensions, setAuthoringReact} from 'appConfig';
+import {registerAuthoringReactFields} from 'apps/authoring-react/fields/register-fields';
+import {registerAuthoringReactWidgets, authoringReactWidgetsExtension} from 'apps/authoring-react/manage-widget-registration';
+import {unregisterInternalExtension} from 'core/helpers/register-internal-extension';
+import {trimStartExact} from 'core/helpers/utils';
+import {flatMap} from 'lodash';
+import ng from 'core/services/ng';
+
+export const switchAuthoring = (url: string) => {
```

Comment:
`maybeEnableAuthoringReact` would be a better name. Or add a comment saying that in certain conditions we're enabling react based authoring